luigi2.h

```diff
@@ -3099,9 +3103,15 @@ int _UITableMessage(UIElement *element, UIMessage message, int di, void *dp) {
 		}
 	} else if (message == UI_MSG_LAYOUT) {
 		UIRectangle scrollBarBounds = element->bounds;
-		scrollBarBounds.l = scrollBarBounds.r - UI_SIZE_SCROLL_BAR * element->window->scale;
 		table->vScroll->maximum = table->itemCount * UI_SIZE_TABLE_ROW * element->window->scale;
 		table->vScroll->page = UI_RECT_HEIGHT(element->bounds) - UI_SIZE_TABLE_HEADER * table->e.window->scale;
+
+		if (table->vScroll->page < table->vScroll->maximum) {
+			scrollBarBounds.l = scrollBarBounds.r - UI_SIZE_SCROLL_BAR * element->window->scale;
+		} else {
+			scrollBarBounds.l = scrollBarBounds.r * element->window->scale;
```

Comment:
The `* element->window->scale` is not needed here.

Reply:
No it is needed, i test it. Set the `scale=2` for e.g. in the `gf2_config.ini` and you'll see.